### PR TITLE
Feature/input sad paths

### PR DESCRIPTION
**Pull request by leighlars**
### Type of change made:
- [X] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
- [X] Refactoring
- [ ] Planning

### Detailed Description
All tests pass in repo files.
Trending related html/css methods are removed as we fulfilled iteration 5 by displaying the friends step counts on the dashboard. 
Sleep methods are implemented. 

### Why is this change required? What problem does it solve?
We want to turn in a quality project and not stretch / diminish quality. We solved everything that was on Iterations 1-5. 

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Yes, just time lengthy.

### Files modified:
- [X] index.html
- [X] .scss // file name(s):
- [X] classes // file name(s): Sleep, User, UserRepo, ActivityRepo
- [X] test // files name(s): SleepRepo, User, UserRepo, ActivityRepo
- [ ] scripts.js
- [ ] Other files:

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -17,15 +17,15 @@ class User {
   }
 
   checkUserId(user) {
-    return typeof user === 'number' ? user : Date.now();
+    return typeof user === "number" ? user : Date.now();
   }
 
   checkName(user) {
-    return typeof user === 'string' ? user : "guest";
+    return typeof user === "string" ? user : "guest";
```
Comment (Leasw144) [inline src/User.js:24]: Thanks for getting those quotes, in hindsight we should have normed on this

File: test/Sleep-Repository-test.js [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
 
-describe('SleepRepository', function() {
+describe("SleepRepository", function () {
   let sleep1, sleep2, sleep3, mockRawData, mockSleepRepo, todaysDate;
   beforeEach(() => {
-    sleep1 = {
+    sleep1 =  {
       "userID": 1,
-      "date": "2019/06/15",
+      "date": "2019/06/17",
```
Comment (Leasw144) [inline test/Sleep-Repository-test.js:10]: What was the reasoning for changing the year?

**Comment by Leasw144** [comment]
Awesome job

File: src/ActivityRepository.js [hunk 3]
```diff
@@ -13,11 +13,12 @@ class ActivityRepository {
...
-    this.individualEntryRecords.filter(record => {
-      return record.date === date; 
-    })
-    return Math.round(this.getStepsForToday(date) * user.strideLength / 5280).toFixed(1); 
+  getUsersMilesForDay(user, date) {
+    if (this.getStepsForToday(date) > 0) {
+      return Number((this.getStepsForToday(date) * user.strideLength / 5280).toFixed(1)); 
+    } else {
+      return 0;
+    }
```
Comment (NakiNorton) [inline src/ActivityRepository.js:21]: Great use of a conditional and `Number` to ensure valid input and correct data type.

File: test/Sleep-Repository-test.js (hunk 2 above)
Comment (leighlars) [reply test/Sleep-Repository-test.js:10]: I was checking to see it's behavior in a sad path

**Pull request merged**